src/main/java/org/reaktivity/nukleus/http2/internal/types/stream/Http2SettingsFW.java

```diff
@@ -123,26 +128,16 @@ public long maxHeaderListSize()
 
     public long settings(int key)
     {
-        int payloadLength = payloadLength();
-        int noSettings = payloadLength/6;
-        if (noSettings > 0)
+        long[] value = new long[] { -1L };
+
+        listFW.forEach(x ->
         {
-            for (int i = 0; i < noSettings; i++)
+            if (x.id() == key)
             {
-                int got = buffer().getShort(offset() + PAYLOAD_OFFSET + 6 * i, BIG_ENDIAN);
-                if (key == got)
-                {
-                    return buffer().getInt(offset() + PAYLOAD_OFFSET + 6 * i + 2, BIG_ENDIAN);
-                }
+                value[0] = x.value();
             }
-        }
-
-        return -1L;
-    }
-
-    public DirectBuffer payload()
-    {
-        return payloadRO;
+        });
```

Comment:
Perhaps we are missing `Stream`-like utility methods on `ListFW`, as this inline lambda would be better specified as `listFW.stream().filter(x -> x.id() == key).findFirst()`.
Please file an issue on `maven-nukleus-plugin` to request that capability on the generated `ListFW` flyweight.